### PR TITLE
driver/Alazar Ats9360 ADC - cleaned up

### PR DESCRIPTION
**Pull request by jenshnielsen**
Reverted some commits that did not belong on #400 and merged with master

### CODE REVIEW

**Comment by jenshnielsen** [comment]
@nataliejpg We should think about getting this into QCoDes soon

**Comment by nataliejpg** [comment]
@jenshnielsen Could we remove it returning raw first or at least make it optional?

File: qcodes/instrument_drivers/AlazarTech/ATS.py [hunk 1]
```diff
@@ -396,10 +412,26 @@ def config(self, clock_source=None, sample_rate=None, clock_edge=None,
                              external_trigger_range)
         self._set_if_present('trigger_delay', trigger_delay)
         self._set_if_present('timeout_ticks', timeout_ticks)
+        self._set_if_present('aux_io_mode', aux_io_mode)
+        self._set_if_present('aux_io_param', aux_io_param)
         # endregion
 
+        # handle that external clock and internal clock uses
+        # two different ways of setting the sample rate.
+        # We use the matching one and make the order one
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/AlazarTech/ATS.py]: mark the other one

File: qcodes/instrument_drivers/AlazarTech/acq_controllers/ATS9360Controller.py [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+        self._update_samples_per_record(sample_rate, int_time, value)
+
+    def _update_num_avg(self, value: int, **kwargs):
+        if not isinstance(value, int) or value < 1:
+            raise ValueError('number of averages must be a positive integer')
+
+
+        if self.acquisition._average_records:
+            self.records_per_buffer._save_val(value)
+            self.buffers_per_acquisition._save_val(1)
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/AlazarTech/acq_controllers/ATS9360Controller.py]: Needs to change

File: qcodes/instrument_drivers/AlazarTech/acq_controllers/ATS9360Controller.py [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+        recordA = self._to_volts(recordA)
+
+        # do demodulation
+        if self.demod_freqs.get_num_demods():
+            magA, phaseA = self._fit(recordA)
+            if self._integrate_samples:
+                magA = np.mean(magA, axis=-1)
+                phaseA = np.mean(phaseA, axis=-1)
+
+        unpacked = []
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/AlazarTech/acq_controllers/ATS9360Controller.py]: only if raw is wanted

**Comment by sohailc** [comment]
Hi guys, what is the status of this driver?

**Comment by jenshnielsen** [comment]
This is replaced by #816

**Pull request closed without merging**